src/Neo/SmartContract/Native/ContractMethodAttribute.cs

```diff
@@ -23,12 +23,19 @@ internal class ContractMethodAttribute : Attribute
         public long CpuFee { get; init; }
         public long StorageFee { get; init; }
         public Hardfork? ActiveIn { get; init; } = null;
+        public Hardfork? DeprecatedIn { get; init; } = null;
```

Comment:
BTW, this is a part of #3210. Add a reference to the PR description, please.

Reply:
Also, we need the same functionality to be implemented for events. It's not hard to extend it, read the original issue.

Reply:
This pr is a hotpatch for the hf issues, i prefer to keep it as simple as possible.

Reply:
Ref. https://github.com/neo-project/neo/issues/3210#issuecomment-2114292377 as TODO.

Reply:
Rename to `ActiveTil`?